### PR TITLE
Clean up for Loading state and tests

### PR DESCRIPTION
**Pull request by solid-akb**
## Description
Was going to be to fix the bug but after deprioritizing still wanted to get some cleanup code in there. 
## Changes
There was some logic and nested ternaries that I think may have caused some unintended loading states. This PR cleans that up. Also there were a few tests that had the wrong props being sent in that have been fixed. 
## Testing
All tests pass.
## Commit checklist

- [ ] All acceptance criteria are met.
- [ ] Includes tests to ensure functionality and prevent regressions.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.

## Screenshots/captures
Mini snippet of logic below: For example in line 159, if page was still loading it would go to the second return in the ternary even if there was no data. 
image

### CODE REVIEW

File: components/resourceDetails/resourceSharing/agentAccessTable/index.jsx [hunk 1]
```diff
@@ -116,47 +115,28 @@ export default function AgentAccessTable({ type, loading, setLoading }) {
...
-            <AgentsSearchBar handleFilterChange={handleFilterChange} />
-          </>
-        )}
-        {loading && (
-          <Container variant="empty">
-            <CircularProgress />
-          </Container>
-        )}
-        {!loading && tablePermissions.length ? (
+  const tableContents = () => {
```
Comment (ThisIsMissEm) [inline components/resourceDetails/resourceSharing/agentAccessTable/index.jsx:120]: would be better to make this an explicit component, but keep it in the same file, that way you've props you can pass to it, and it won't be redeclared each render.

File: components/resourceDetails/resourceSharing/agentAccessTable/index.jsx [hunk 2]
```diff
@@ -191,53 +171,88 @@ export default function AgentAccessTable({ type, loading, setLoading }) {
...
+          {/* TODO: Uncomment to reintroduce tabs */}
+          {/* <Tabs */}
+          {/*  handleTabChange={handleTabChange} */}
+          {/*  selectedTabValue={selectedTabValue} */}
+          {/*  tabs={[{ label: "All", value: "all", testid: TESTCAFE_ID_TAB_ALL },  { label: "People", value: "agent", testid: TESTCAFE_ID_TAB_PEOPLE }, { label: "Groups", value: "group", testid: TESTCAFE_ID_TAB_GROUPS } }]} */}
+          {/* /> */}
+          <AgentsSearchBar handleFilterChange={handleFilterChange} />
+        </>
+      )}
+      {tableContents()}
```
Comment (ThisIsMissEm) [inline components/resourceDetails/resourceSharing/agentAccessTable/index.jsx:255]: Then here you'd have `<PermissionsTable ... />` with the various props it needs

**Review by ThisIsMissEm — APPROVED** [review]
Looks overall good, but one comment structure wise

File: components/resourceDetails/resourceSharing/agentAccessTable/index.jsx (hunk 1 above)
Comment (solid-akb) [reply components/resourceDetails/resourceSharing/agentAccessTable/index.jsx:120]: Awesome - makes sense. Thanks!

**Pull request merged**